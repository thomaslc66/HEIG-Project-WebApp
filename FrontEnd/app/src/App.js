/* eslint-disable react/jsx-filename-extension */
import React, { Component } from 'react';
import { Alert, Intent, Position, Toaster } from '@blueprintjs/core';
import { connect } from 'react-redux';
import Home from './container/home/Home';
import { axiosPostQuery } from './utils/axiosQuery';
import {
  deleteContactFromResults,
  setContactToDelete,
  resetContactToDelete,
  handleMoveCancel,
} from './redux/actions/contact-actions';
import { contact } from './utils/routes';
import SideMenu from 'react-sidemenu';

class App extends Component {
  state = {
    isProgramSelectorOpen: false,
  };

  /** *********************************************************************
   *
   * Method that can render a toaster from @blueprint/core on any pages.
   *
   *
   * ******************************************************************* */
  renderToaster = (intent, message) => {
    this.toaster.show({
      intent,
      message,
    });
  };

  /** *********************************************************************
   *
   * Methods that can render a alert when user want to delete a contact
   *
   * ********************************************************************** */
  handleMoveConfirm = () => {
    this.deleteContact();
  };

  onProgramConfirm = () => {
    this.setState({ isProgramSelectorOpen: false });
  };

  onProgramOpen = () => {
    this.setState({ isProgramSelectorOpen: true });
  };

  /** *********************************************************************
   * @name deleteContact
   * @description Methods that can render a alert when user
   * want to delete a contact. The method delete the user from the database
   *
   * ********************************************************************** */
  deleteContact = () => {
    const {
      onDeleteContactFromResults,
      onResetContactToDelete,
      contactToDelete,
    } = this.props;

    // Send query to the backend
    axiosPostQuery(contact.delete, { contactID: contactToDelete._id })
      .then(response => {
        if (response.status === 200) {
          this.toaster.show({
            intent: Intent.SUCCESS,
            message: `${contactToDelete.name} a été effacé`,
          });

          onDeleteContactFromResults(contactToDelete);
          onResetContactToDelete();
        }
      })
      .catch(() => {
        this.toaster.show({
          intent: Intent.DANGER,
          message: `${contactToDelete.name} n'a pas pu être effacé`,
        });
        onResetContactToDelete();
      });
  };

  render() {
    const {
      isOpen,
      contactToDelete,
      canEscapeKeyCancel,
      onSetContactToDelete,
      onHandleMoveCancel,
    } = this.props;

    return (
      <div className="App">
        <Home
          renderToaster={this.renderToaster}
          deleteContact={onSetContactToDelete}
        />
        <Toaster
          position={Position.BOTTOM}
          ref={element => {
            this.toaster = element;
          }}
        />
        <Alert
          cancelButtonText="Annuler"
          confirmButtonText="Confirmer"
          icon="trash"
          intent={Intent.DANGER}
          isOpen={isOpen}
          canEscapeKeyCancel={canEscapeKeyCancel}
          onCancel={onHandleMoveCancel}
          onConfirm={this.handleMoveConfirm}>
          <p>
            Êtes-vous sur de vouloir effacer <b>{contactToDelete.name}</b>{' '}
            définitivement ? Cette opération est irréversible !
          </p>
        </Alert>
      </div>
    );
  }
}

/* Redux Management */
const mapStateToProps = state => ({
  contactToDelete: state.contact.contact,
  canEscapeKeyCancel: state.contact.canEscapeKeyCancel,
  canOutsideClickCancel: state.contact.canOutsideClickCancel,
  isOpen: state.contact.isOpen,
});

const mapActionToProps = {
  onSetContactToDelete: setContactToDelete,
  onHandleMoveCancel: handleMoveCancel,
  onResetContactToDelete: resetContactToDelete,
  onDeleteContactFromResults: deleteContactFromResults,
};

/* Component Export */
export default connect(
  mapStateToProps,
  mapActionToProps
)(App);
