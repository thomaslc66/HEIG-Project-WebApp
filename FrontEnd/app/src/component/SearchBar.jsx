import React, { Component } from 'react';
import AutoSuggest from 'react-autosuggest';
import Axios from 'axios';
import Suggestions from './Suggestions';

const getSuggestionValue = suggestion => suggestion.name;

const renderSuggestion = suggestion => (
  <div className='resultat'>{suggestion.name}</div>
);

class SearchBar extends Component {
  constructor() {
    super();
    this.state = {
      value: '',
      query: '',
      suggestions: [],
      results: []
    };
  }

  getInfo = () => {
    const newValue = this.state.query;
    Axios.get(`http://localhost:4000/suppliers/tag/${newValue}`).then(
      response => {
        console.log(response.data);
        this.setState({
          results: response.data
        });
      }
    );
  };

  handleInputChange = () => {
    this.setState(
      {
        query: this.search.value
      },
      () => {
        if (this.state.query && this.state.query.length >= 1) {
          this.getInfo();
        } else if (!this.search.query) {
          this.setState({ results: [] });
        }
      }
    );
  };

  render() {
    return (
      <form>
        <input
          placeholder='Search for...'
          ref={input => (this.search = input)}
          onChange={this.handleInputChange}
        />
        <Suggestions results={this.state.results} />
      </form>
    );
  }
}

export default SearchBar;
