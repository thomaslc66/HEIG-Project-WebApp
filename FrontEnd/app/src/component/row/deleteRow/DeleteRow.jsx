import React, { Component } from 'react';
import './DeleteRow.css';
import { MDBIcon } from 'mdbreact';

class DeleteRow extends Component {
  render() {
    const { disabled, index } = this.props;
    if (disabled) {
      return <td className="align-vertical">{index + 1}</td>;
    } else {
      return (
        <td className="align-vertical">
          <MDBIcon
            index={index}
            icon="trash-alt"
            className="text-danger "
            onClick={() => this.props.deleteARow()}
          />
        </td>
      );
    }
  }
}

export default DeleteRow;
