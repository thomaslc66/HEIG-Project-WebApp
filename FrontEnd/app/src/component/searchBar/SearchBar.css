/* Top page bar */
.searchBar {
  display: inline-flex;
}

.searchIcon {
  padding: 15px;
  border: 1px solid #aaa;
  color: white;
  border-bottom-right-radius: 5px;
  border-top-right-radius: 5px;
  background-color: #5a95f5;
}

.eraseIcon {
  padding: 15px;
  border: 1px solid #aaa;
  color: white;
  border-bottom-right-radius: 5px;
  border-top-right-radius: 5px;
  background-color: #5a95f5;
  cursor: pointer;
}

.eraseIcon:hover {
  color: #f98400;
}

/* Auto Suggest info */
.react-autosuggest__container {
  position: relative;
}

.react-autosuggest__input {
  width: 360px;
  height: 50px;
  padding: 0px 12px;
  font-family: Roboto;
  font-weight: 350;
  font-size: 18px;
  border: 1px solid #aaa;
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 5px;
  border-top-right-radius: 0;
  border-top-left-radius: 5px;
}

/* Auto Suggest Feature CSS Stlye*/
.react-autosuggest__input:focus {
  outline: none;
}

.react-autosuggest__container--open .react-autosuggest__input {
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.react-autosuggest__suggestions-container {
  display: none;
}

.react-autosuggest__container--open .react-autosuggest__suggestions-container {
  display: block;
  position: absolute;
  top: 49px;
  width: 360px;
  border: 1px solid #aaa;
  background-color: #fff;
  font-family: Roboto;
  font-weight: 300;
  font-size: 15px;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  z-index: 2;
  text-align: left;
}

.react-autosuggest__suggestions-list {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.react-autosuggest__suggestion {
  cursor: pointer;
  padding: 0px 10px;
}

.react-autosuggest__suggestion:not(:first-child) {
  border-top: 1px solid #ddd;
}

.react-autosuggest__suggestion--focused {
  background-color: #5a95f5;
  color: rgb(19, 18, 18);
}

.suggestion-content {
  display: flex;
  align-items: center;
  background-repeat: no-repeat;
}

.name {
  line-height: 45px;
}

.highlight {
  color: #5a95f5;
  font-weight: bold;
}

.react-autosuggest__suggestion--highlighted {
  background-color: rgb(241, 237, 237);
}

.react-autosuggest__suggestion--highlighted .highligh {
  color: #120000;
}

.react-autosuggest__suggestion--focused .highlight {
  color: #120000;
}
