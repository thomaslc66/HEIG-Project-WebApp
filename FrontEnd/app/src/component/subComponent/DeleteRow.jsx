import React, { Component } from 'react';
import { MDBIcon } from 'mdbreact';

class DeleteRow extends Component {
	state = {};
	render() {
		const { status, index, indexPath } = this.props;
		if (status) {
			return <td>{index + 1}</td>;
		} else {
			return (
				<td>
					<MDBIcon
						index={index}
						icon="trash-alt"
						className="text-danger"
						onClick={e => this.props.deleteARow(e, indexPath)}
					/>
				</td>
			);
		}
	}
}

export default DeleteRow;
