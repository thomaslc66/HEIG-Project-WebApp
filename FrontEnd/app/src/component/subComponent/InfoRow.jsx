import React, { Component } from 'react';
import { MDBIcon } from 'mdbreact';
import copy from 'copy-to-clipboard';
import { Intent } from '@blueprintjs/core';
import AutosizeInput from 'react-input-autosize';

class InfoRow extends Component {
	state = {
		isHovered: false
	};

	copyText = value => {
		if (value) {
			copy(value);
			this.props.renderToaster(Intent.SUCCESS, `Copie réussie - plus qu'a coller`);
		} else {
			this.props.renderToaster(Intent.WARNING, `Rien à copier`);
		}
	};

	handleHover = () => {
		this.setState(prevState => ({
			isHovered: !prevState.isHovered
		}));
	};

	render() {
		const { color, icon, placeholder, status, text, name, error } = this.props;
		return (
			<tr className="tr_info_style">
				<td className="td_info_contact">
					<MDBIcon icon={icon} size="lg" className={`icon text-${color}`} />
					<AutosizeInput
						className="list_input"
						placeholder={placeholder}
						disabled={status}
						value={text}
						name={name}
						onChange={e => this.props.onChange(e)}
					/>
					<MDBIcon
						far
						icon="clone"
						className={this.state.isHovered ? 'icon_clone_hover' : 'icon_clone text-muted'}
						onClick={() => this.copyText(this.props.text)}
						onMouseEnter={this.handleHover}
						onMouseLeave={this.handleHover}
					/>
					<span className="info_error">{error}</span>
				</td>
			</tr>
		);
	}
}

export default InfoRow;
