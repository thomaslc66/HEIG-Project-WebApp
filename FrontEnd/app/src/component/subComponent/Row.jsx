import React, { Component } from 'react';
import AutosizeInput from 'react-input-autosize';

class Row extends Component {
  render() {
    console.log(this.props);
    const {
      placeholder,
      onChange,
      status,
      text,
      name,
      error,
      ...restProps
    } = this.props;
    return (
      <React.Fragment>
        <AutosizeInput
          {...restProps}
          className="list_input"
          placeholder={placeholder}
          disabled={status}
          value={text}
          name={name}
          onChange={onChange}
        />
        <span className="info_error">{error}</span>
      </React.Fragment>
    );
  }
}

export default Row;
