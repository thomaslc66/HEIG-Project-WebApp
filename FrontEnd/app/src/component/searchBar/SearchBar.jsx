import React, { Component } from 'react';
import './SearchBar.css';
import { MDBIcon } from 'mdbreact';
import { connect } from 'react-redux';
import {
  runSearch,
  onChange,
  onSuggestionsClearRequested,
  onSuggestionsFetchRequested,
  onEraseSearchValue,
} from '../../redux/actions/contact-actions';
import Autosuggest from 'react-autosuggest';
import AutosuggestHighlightMatch from 'autosuggest-highlight/match';
import AutosuggestHighlightParse from 'autosuggest-highlight/parse';
import { SEARCH } from '../../utils/string';
import { Intent } from '@blueprintjs/core';

function renderSuggestion({ name, ...suggestion }, { query }) {
  const matches = AutosuggestHighlightMatch(name, query);
  const parts = AutosuggestHighlightParse(name, matches);
  return (
    <span className={'suggestion-content '}>
      <span className="name">
        {parts.map((part, index) => {
          const className = part.highlight ? 'highlight' : null;
          return (
            <span className={className} key={index}>
              {part.text}
            </span>
          );
        })}
      </span>
    </span>
  );
}

class SearchBar extends Component {
  getSuggestionValue = suggestion => suggestion.name;

  onSuggestionSelected = (event, { suggestionValue }) => {
    if (this.props.isNewContactOn) {
      this.props.renderToaster(
        Intent.WARNING,
        "Ajout de contact en cours. Merci d'annuler ou terminer"
      );
    } else if (this.props.isEditingOn) {
      this.props.renderToaster(
        Intent.WARNING,
        'Edition du contact en cours. Merci de terminer'
      );
    } else {
      this.props.onRunSearch(suggestionValue);
    }
  };

  eraseSearchValue = () => {
    if (this.props.isEditingOn) {
      this.props.renderToaster(
        Intent.WARNING,
        'Edition du contact en cours. Merci de terminer'
      );
    } else {
      this.props.eraseSearchValue();
    }
  };

  onKeyPress = e => {
    if (e.key === 'Enter') {
      if (this.props.isNewContactOn) {
        this.props.renderToaster(
          Intent.WARNING,
          "Ajout de contact en cours. Merci d'annuler ou terminer"
        );
      } else if (this.props.isEditingOn) {
        this.props.renderToaster(
          Intent.WARNING,
          'Edition du contact en cours. Merci de terminer'
        );
      } else {
        this.props.onRunSearch(e.target.value);
      }
    }
  };

  render() {
    const {
      value,
      suggestions,
      onChange,
      onSuggestionsFetchRequested,
      onSuggestionsClearRequested,
    } = this.props;

    const inputProps = {
      placeholder: SEARCH,
      value,
      onChange,
      onKeyPress: this.onKeyPress,
    };

    return (
      <div className="searchBar">
        <Autosuggest
          suggestions={suggestions}
          onSuggestionsFetchRequested={onSuggestionsFetchRequested}
          onSuggestionsClearRequested={onSuggestionsClearRequested}
          getSuggestionValue={this.getSuggestionValue}
          renderSuggestion={renderSuggestion}
          inputProps={inputProps}
          onSuggestionSelected={this.onSuggestionSelected}
        />
        {value ? (
          <MDBIcon
            className="eraseIcon"
            icon="times"
            onClick={this.eraseSearchValue}
          />
        ) : (
          <MDBIcon className="searchIcon" icon="search" />
        )}
      </div>
    );
  }
}

/* Redux Management */
const mapStateToProps = state => ({
  allContactName: state.contact.allContactName,
  value: state.contact.value,
  suggestions: state.contact.suggestions,
  isNewContactOn: state.contact.isNewContactOn,
  isEditingOn: state.contact.isEditingOn,
});

const mapActionToProps = {
  onRunSearch: runSearch,
  onChange: onChange,
  onSuggestionsFetchRequested: onSuggestionsFetchRequested,
  onSuggestionsClearRequested: onSuggestionsClearRequested,
  eraseSearchValue: onEraseSearchValue,
};

/* Component Export */
export default connect(
  mapStateToProps,
  mapActionToProps
)(SearchBar);
