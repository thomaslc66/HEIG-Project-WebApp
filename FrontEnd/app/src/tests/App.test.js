import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
import App from '../App';
import Row from '../component/row/Row';
import InfoRow from '../component/row/InfoRow/InfoRow';
import DeleteRow from '../component/row/DeleteRow/DeleteRow';

import SearchBar from '../component/searchBar/SearchBar';

Enzyme.configure({ adapter: new Adapter() });

it('Render InfoRow Component without crashing', () => {
  const RowComponent = renderer.create(<InfoRow />).toJSON();
  expect(RowComponent).toMatchSnapshot();
});

it('Render correctly Row component', () => {
  const RowComponent = renderer.create(<Row />).toJSON();
  expect(RowComponent).toMatchSnapshot();
});

it('Render correctly SearchBar component', () => {
  const RowComponent = renderer.create(<DeleteRow />).toJSON();
  expect(RowComponent).toMatchSnapshot();
});
