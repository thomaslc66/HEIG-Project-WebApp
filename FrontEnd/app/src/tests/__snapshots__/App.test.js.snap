// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render InfoRow Component without crashing 1`] = `
<tr
  className="tr_info_style"
>
  <td
    className="td_info_contact"
  >
    <i
      className="fa fa-circle fa-lg icon text-undefined"
    />
    <div
      className="list_input"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <input
        onChange={[Function]}
        style={
          Object {
            "boxSizing": "content-box",
            "width": "1px",
          }
        }
      />
      <div
        style={
          Object {
            "height": 0,
            "left": 0,
            "overflow": "scroll",
            "position": "absolute",
            "top": 0,
            "visibility": "hidden",
            "whiteSpace": "pre",
          }
        }
      >
        
      </div>
    </div>
    <i
      className="far fa-clone icon_clone text-muted"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
    <span
      className="info_error"
    />
  </td>
</tr>
`;

exports[`Render correctly Row component 1`] = `
Array [
  <div
    className="list_input"
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <input
      style={
        Object {
          "boxSizing": "content-box",
          "width": "1px",
        }
      }
    />
    <div
      style={
        Object {
          "height": 0,
          "left": 0,
          "overflow": "scroll",
          "position": "absolute",
          "top": 0,
          "visibility": "hidden",
          "whiteSpace": "pre",
        }
      }
    >
      
    </div>
  </div>,
  <span
    className="info_error"
  />,
]
`;
